.github/license-lint-config.yml

```diff
@@ -0,0 +1,68 @@
+unrestricted_licenses:
+  - Apache-2.0
+  - BSD-2-Clause
+  - BSD-3-Clause
+  - 0BSD
+  - MIT
+
+restricted_licenses:
+  - bzip2-1.0.5
+  - AGPL-1.0
+  - AGPL-3.0
+  - BSD-2-Clause-FreeBSD
+  - BSD-2-Clause-NetBSD
+  - eCos-2.0
+  - GFDL-1.1
+  - GFDL-1.2
+  - GFDL-1.3
+  - GPL-1.0-only
+  - GPL-1.0-or-later
+  - GPL-2.0+
+  - GPL-2.0-only
+  - GPL-2.0-or-later
+  - GPL-2.0-with-autoconf-exception
+  - GPL-2.0-with-bison-exception
+  - GPL-2.0-with-classpath-exception
+  - GPL-2.0-with-GCC-exception
+  - GPL-2.0-with-font-exception
+  - GPL-3.0-only
+  - GPL-3.0-or-later
+  - GPL-3.0+
+  - GPL-3.0-with-autoconf-exception
+  - GPL-3.0-with-GCC-exception
+  - LGPL-2.0-only
+  - LGPL-2.0+
+  - LGPL-2.0-or-later
+  - LGPL-2.1-only
+  - LGPL-2.1-or-later
+  - LGPL-3.0-only
+  - LGPL-3.0-or-later
+  - LGPL-3.0+
+  - NPL-1.0
+  - NPL-1.1
+  - Nunit
+  - OSL-1.0
+  - OSL-1.1
+  - OSL-2.0
+  - OSL-2.1
+  - OSL-3.0
+  - QPL-1.0
+  - Sleepycat
+  - StandardML-NJ
+  - wxWindows
+
+allowlisted_modules:
```

Comment:
why do we need to specifically allow these?

Reply:
some of these have some text modifications that lowers the confidence score below a threshold. ref:

https://github.com/istio/tools/blob/master/cmd/license-lint/analysis.go#L75-L81 

some have license files that are not where they should be IIUC

Reply:
What is the mechanism for detecting license changes in allowlisted modules?

IMO an unexpected license location/name is not a reason to adopt extra risk that a project changes its license.

Does this project allow us to manually specify license locations? Are there other linters that would more readily deal with these edge cases?

Reply:
Addressing the first question as a separate comment as to why we are changing the underlying tool used to lint in the GH wf.

> Does this project allow us to manually specify license locations? Are there other linters that would more readily deal with these edge cases?

Unfortunately, the k8s script doesn't do that for us either; they hard code the location for k8s.io for instance and we can probably take the same approach where needed.